tests/bud.bats

```diff
@@ -117,6 +117,61 @@ symlink(subdir)"
   expect_output --substring $(realpath "${TESTSDIR}/bud/dockerignore3/.dockerignore")
 }
 
+# Following test must fail since we are trying to run linux/arm64 on linux/amd64
+# Issue: https://github.com/containers/buildah/issues/3712
+@test "build-with-inline-platform" {
+  # Host arch
+  run_buildah info --format '{{.host.arch}}'
+  myarch="$output"
+  otherarch="arm64"
+
+  # just make sure that other arch is not equivalent to host arch
+  if [[ "$otherarch" == "$myarch" ]]; then
+    otherarch="amd64"
+  fi
+  # ...create a Containerfile with --platform=linux/$otherarch
+  cat > ${TESTSDIR}/bud/platform/Dockerfile << _EOF
+FROM --platform=linux/${otherarch} alpine
+RUN uname -m
+_EOF
+
+  run_buildah '?' build --signature-policy ${TESTSDIR}/policy.json -t test ${TESTSDIR}/bud/platform
+  if [[ $status -eq 0 ]]; then
+    # Build succeeded: we must have qemu-user-static installed. Confirm that
+    # the 'uname' emits what we expect
+    expect_output --from="${lines[-1]}" "$otherarch"
+    run_buildah inspect --format '{{ .OCIv1.Architecture }}' test
+    expect_output --substring "$otherarch"
+  else
+    # Build failed: we DO NOT have qemu-user-static installed.
+    expect_output --substring "format error"
+  fi
+}
+
+# Following test must pass since we want to tag image as host arch
+# Test for use-case described here: https://github.com/containers/buildah/issues/3261
+@test "build-with-inline-platform-amd-but-tag-as-arm" {
+  # Host arch
+  run_buildah info --format '{{.host.arch}}'
+  myarch="$output"
+  targetarch="arm64"
+
+  if [[ "$targetArch" == "$myarch" ]]; then
+    targetarch="amd64"
+  fi
+
+  cat > ${TESTSDIR}/bud/platform/Dockerfile << _EOF
+FROM --platform=linux/${myarch} alpine
+RUN uname -m
+_EOF
+
+  # Tries building image where baseImage has --platform=linux/HostArch
+  run_buildah '?' build --platform linux/${targetarch} --signature-policy ${TESTSDIR}/policy.json -t test ${TESTSDIR}/bud/platform
+  run_buildah inspect --format '{{ .OCIv1.Architecture }}' test
```

Comment:
That would fail if no qemu isn't installed.  Don't we need the same checks as in the test abov?

Reply:
@vrothberg No this should not fail since we are only tagging as different `arch` the image which we are actually pulling is specifically set to match `host` arch. I am just creating `dockerfile` dynamically for specific host.

Reply:
The `RUN` instruction is still there, and we ignore the exit code.  If the build fails, there's no tagged image and then the inspect should fail.

Reply:
@vrothberg I'm confused a bit I guess. But the idea is if we pull the image which is equivalent to `hostarch` `RUN` should never fail.

Reply:
We build for a non-local platform, so `build` could fail if qemu isn't installed, couldn't it?  If it's intended to succeed, I am confused why we ignore the exist code, see `?`.

Reply:
@vrothberg I have removed the `?` so if build fails the test will break. I think extra `?` was added cause i copied this test from the first one.

Reply:
The `--platform` on the FROM instruction implicitly overrides the `--platform` we get at the command line, and this looks like it's attempting to verify that the build succeeds when the command line `--platform` points to a foreign architecture, but `FROM --platform` specifies the local one.  It'll succeed either way if emulation is enabled, though, which is probably not what we want.

Reply:
Ah, and that's why it checks on the next line.

Reply:
@nalind That's correct.

Reply:
> The --platform on the FROM instruction implicitly overrides the --platform we get at the command line, and this looks like it's attempting to verify that the build succeeds when the command line --platform points to a foreign architecture, but FROM --platform specifies the local one. It'll succeed either way if emulation is enabled, though, which is probably not what we want.

Is the behavior that the Dockerfile instructions overrides the CLI inherited from Docker or buildkit?  My expectation is that when I say `--platform foo` on the CLI that I get platform foo.

Reply:
@vrothberg Yes `buidlkit` supports that. The reason is so users could do `multi-platform` builds otherwise the variables which we set with cli `--platform` like `BUILDPLATFORM` and `TARGETPLATFORM` would not be a lot of useful to end users if they can't override base images.

There is more discussed here: https://github.com/containers/buildah/issues/3261 and https://docs.docker.com/buildx/working-with-buildx/#build-multi-platform-images

Tryout example with `buildkit`:

```dockerfile
FROM --platform=linux/amd64 alpine
RUN uname -m
```

But build this docker file using `--platform` different than one set in `Dockerfile`.

```console
sudo docker buildx build --platform linux/arm64 -t test .
```

Now on inspecting this image `sudo docker inspect test`  you can actually see that image for overridden by `inline` `linux/amd64` platform but still tagged as `linux/arm64`.